bundles/org.openhab.binding.sony/src/main/java/org/openhab/binding/sony/internal/net/NetUtil.java

```diff
@@ -0,0 +1,335 @@
+                if (StringUtils.isEmpty(mimeType)) {
+                    // probably a 'content' header of value 'Content-Type: image/png' instead
+                    mimeType = resp.getResponseHeader("content");
+                    final int idx = mimeType.indexOf(":");
+                    if (idx >= 0) {
+                        mimeType = mimeType.substring(idx + 1).trim();
+                    }
+                }
+            }
+        }
+
+        return iconData == null ? null : new RawType(iconData, mimeType);
+    }
+    /**
+     * Send a request to the specified ipaddress/port using a socket connection. Any results will be sent back via the
+     * callback.
+     *
+     * @param ipAddress the non-null, non-empty ip address
+     * @param port the port
+     * @param request the non-null, non-empty request
+     * @param callback the non-null callback
+     * @throws IOException if an IO exception occurs sending the request
+     */
+    public static void sendSocketRequest(final String ipAddress, final int port, final String request,
+            final SocketSessionListener callback) throws IOException {
+        Validate.notEmpty(ipAddress, "ipAddress cannot be empty");
+        Validate.notEmpty(request, "request cannot be empty");
+        Objects.requireNonNull(callback, "callback cannot be null");
+
+        final Socket socket = new Socket();
+        try {
+            socket.setSoTimeout(10000);
+            socket.connect(new InetSocketAddress(ipAddress, port));
+
+            final PrintStream ps = new PrintStream(socket.getOutputStream());
+            final BufferedReader bf = new BufferedReader(new InputStreamReader(socket.getInputStream()));
+
+            ps.println(request + "\n");
+            ps.flush();
+
+            int c;
+            final StringBuilder sb = new StringBuilder(100);
+            while (true) {
+                try {
+                    c = bf.read();
+                    if (c == -1) {
+                        final String str = sb.toString();
+                        callback.responseReceived(str);
+                        break;
+                    }
+                    final char ch = (char) c;
+                    if (ch == '\n') {
+                        final String str = sb.toString();
+                        sb.setLength(0);
+                        if (callback.responseReceived(str)) {
+                            break;
+                        }
+                    }
+                    sb.append(ch);
+                } catch (final SocketTimeoutException e) {
+                    final String str = sb.toString();
+                    callback.responseReceived(str);
+                    break;
+
+                } catch (final IOException e) {
+                    callback.responseException(e);
+                    break;
+                }
+            }
+        } finally {
+            socket.close();
+        }
```

Comment:
Isn't this overly complicated? using `bf.readLine()` would simplify it a bunch.

Reply:
That one actually screwed me up awhile back.  The spec says that a new line is the ending of a message - readline will read up to a new line **or carriage return**.  As I found out, there can be carriage return in the data portion of a message (example: if someone names an input "hdmi1\rbluray" (where the \r is a carriage return on the screen) - so I had to switch to a manual mode to only do new lines